src/view/main_scene_view.cpp

```diff
@@ -3,18 +3,79 @@
 #include "view/main_scene_view.hpp"
 
 MainSceneView::MainSceneView()
-: camera(glm::radians(45.0f), 800 / 600, .1f, 100.0f)
+: View()
 {
-	//todo temporary below
-	camera.setPosition(glm::vec3(.0f, 3.0f, 10.0f));
+	camera = new AboveCamera(glm::radians(45.0f), 800 / 600, .1f, 100.0f, 0.05f);
+	camera->setPosition(glm::vec3(0.0f, 3.0f, 0.0f));
+
+}
+
+void MainSceneView::processEvents(SDL_Event &event)
+{
+	AboveCamera* aboveCamera = (AboveCamera*)camera;
```

Comment:
This line has absolutely no place, we are keeping a pointer to a Camera, and it can point to any camera that is derived from Camera base class. We are using polymorphism here, and because of that all methods that are called on our Camera pointer in view have to have interfaces in base Camera class.